automation_testing/pages/SaveAsComponent.ts

```diff
@@ -0,0 +1,97 @@
+import { expect, Locator, Page } from '@playwright/test';
+import { url } from '../utils/formUrls';
+
+export class SaveAsComponent {
+    readonly page: Page;
+    readonly saveAsBtn: Locator;
+    readonly saveAsOptions: Locator;
+    readonly saveView: Locator;
+    readonly saveLabel: Locator;
+    readonly saveViewBtn: Locator;
+    readonly cancelViewBtn: Locator;
+    readonly viewBtn: Locator;
+    readonly saveAsCheckbox: Locator;
+    readonly tableViews: Locator;
+    readonly editView: Locator;
+    readonly editCheckbox: Locator;
+    readonly closeSaveViewBtn: Locator;
+    readonly closeEditViewBtn: Locator;
+
+    constructor(page: Page) {
+        this.page = page;
+        this.saveAsBtn = page.locator('text=Save As');
+        this.saveAsOptions = page.locator('ul.MuiList-padding');
+        this.saveView = page.locator('//html/body/div[5]/div[3]/div/div');
+        this.saveLabel = page.locator('#label');
+        this.saveViewBtn = page.locator('div[role="presentation"] button:has-text("Save")');
+        this.cancelViewBtn = page.locator('div[role="presentation"] button:has-text("Cancel")');
+        this.viewBtn = page.locator('//*[@id="root"]/div/div/div[1]/div/div[2]/div/span/span[2]/button/span[1]/span');
+        this.saveAsCheckbox = this.saveView.locator('input[type=checkbox]');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/simpleviewinc/sv-mosaic/pullRequest?prid=9498610)
```suggestion
        this.saveAsCheckbox = this.saveView.locator("input[type=checkbox]");
```